tests/primer3/test_primer3_parameters.py

```diff
@@ -18,33 +19,116 @@ def valid_primer3_params() -> Primer3Parameters:
+    mapped_dict = valid_probe_params.to_input_tags()
+    # because `probe_excluded_region` is not given, we do not expect a key in `mapped_dict`
+    assert Primer3InputTag.SEQUENCE_INTERNAL_EXCLUDED_REGION not in mapped_dict
+
+
+def test_probe_param_exclude_regions_construction_valid(
+    valid_probe_params_with_exclude_regions: ProbeParameters,
+) -> None:
+    """Test ProbeParameters class instantiation with valid input for
+    `SEQUENCE_INTERNAL_EXCLUDED_REGION` and ensure `to_input_tags() generates
+    corresponding key/value pair."""
+    assert valid_probe_params_with_exclude_regions.probe_excluded_region == (1, 10)  # expect tuple
+    mapped_dict = valid_probe_params_with_exclude_regions.to_input_tags()
+    assert mapped_dict[Primer3InputTag.SEQUENCE_INTERNAL_EXCLUDED_REGION] == "1,10"  # expect string
+
+
+def test_primer_amplicon_param_construction_raises(
+    valid_primer_amplicon_params: PrimerAndAmpliconParameters,
+) -> None:
+    """Test that PrimerAndAmpliconParameters post_init raises with invalid input."""
     # overriding mypy here to test a case that normally would be caught by mypy
     with pytest.raises(ValueError, match="Primer Max Dinuc Bases must be an even number of bases"):
         # replace will create a new Primer instance with the provided/modified arguments
-        replace(valid_primer3_params, primer_max_dinuc_bases=5)
+        replace(valid_primer_amplicon_params, primer_max_dinuc_bases=5)
     with pytest.raises(TypeError, match="Amplicon sizes and primer sizes must be integers"):
-        replace(valid_primer3_params, amplicon_sizes=MinOptMax(min=200.0, opt=250.0, max=300.0))  # type: ignore
+        replace(
+            valid_primer_amplicon_params,
+            amplicon_sizes=MinOptMax(min=200.0, opt=250.0, max=300.0),  # type: ignore
+        )
     with pytest.raises(TypeError, match="Amplicon sizes and primer sizes must be integers"):
-        replace(valid_primer3_params, primer_sizes=MinOptMax(min=18.0, opt=21.0, max=27.0))  # type: ignore
+        replace(valid_primer_amplicon_params, primer_sizes=MinOptMax(min=18.0, opt=21.0, max=27.0))  # type: ignore
     with pytest.raises(ValueError, match="Min primer GC-clamp must be <= max primer GC-clamp"):
-        replace(valid_primer3_params, gc_clamp=(5, 0))
+        replace(valid_primer_amplicon_params, gc_clamp=(5, 0))
+
+
+def test_primer_probe_param_construction_raises(
+    valid_probe_params: ProbeParameters,
+) -> None:
+    """Test that Primer3Parameters post_init raises with invalid input."""
+    # overriding mypy here to test a case that normally would be caught by mypy
+    with pytest.raises(TypeError, match="Probe sizes must be integers"):
+        # replace will create a new Primer instance with the provided/modified arguments
+        # we use `type: ignore` here to bypass mypy
+        replace(
+            valid_probe_params,
+            probe_sizes=MinOptMax(min=18.1, opt=22.1, max=30.1),  # type: ignore
```

Comment:
**issue** Add the ignore code, and a comment that the the mypy errors are deliberate in the context of this test - we are deliberately passing the wrong type to test our runtime checks